### PR TITLE
[Merged by Bors] - feat: modify analytics service call to new ingest service [bugfix] (VF-3505)

### PR DESCRIPTION
**Pull request by pmvrmc**
**Fixes or implements VF-3505**

### Brief description. What is this change?

We need to change the google-runtime to make the calls go to the new event-ingestion service, modifying the payload to support it

### Implementation details. How do you make this change?

Adapting the analytics client to call the new endpoint with the correct payload

### Setup information

must be used with the other general-runtime PR in order to work properly

### Deployment Notes

must be deployed after the general-runtime PR in order to work properly
we also need to set the `INGEST_WEBHOOK_ENDPOINT` env variable correctly for deployment

### Related PRs

- https://github.com/voiceflow/database-cli/pull/193
- https://github.com/voiceflow/alexa-runtime/pull/290
- https://github.com/voiceflow/general-runtime/pull/314
- https://github.com/voiceflow/creator-api/pull/988
- https://github.com/voiceflow/creator-app/pull/5454

### Checklist

- [ ] this is a breaking change and should publish a new major version
- [ ] appropriate tests have been written

### CODE REVIEW

File: lib/services/dialogflow/lifecycle/es/response.ts [hunk 1]
```diff
@@ -60,13 +60,15 @@ class ResponseManager extends AbstractManager<{ utils: typeof utilsObj }> {
     await state.saveToDb(storage.get<string>(S.USER)!, runtime.getFinalState());
 
     const versionID = runtime.getVersionID();
+    const { projectID } = await runtime.api.getVersion(versionID);
```
Comment (pmvrmc) [inline lib/services/dialogflow/lifecycle/es/response.ts:63]: im not sure this is the best way to pull the projectID, ideas?

File: lib/clients/analytics.ts [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
-import { WebhookResponse } from '../services/dialogflow/types';
 import { GoogleRequest } from '../services/types';
-import { InteractBody, TurnBody } from './ingest-client';
+import { InteractionBody, Payload, TraceBody } from './ingest-client';
 import { AbstractClient } from './utils';
 
-type Payload = SimpleResponse | GoogleRequest | Simple | WebhookResponse;
+const isGoogleRequest = (p: Payload): p is GoogleRequest => (p ? 'type' in p : false);
+
+// type Payload = SimpleResponse | GoogleRequest | Simple | WebhookResponse;
```
Comment (jonahsnider) [inline lib/clients/analytics.ts:13]: this is no longer needed?

File: lib/clients/analytics.ts [hunk 3]
```diff
@@ -1,24 +1,23 @@
...
 import { Config } from '@/types';
 
-import { WebhookResponse } from '../services/dialogflow/types';
 import { GoogleRequest } from '../services/types';
-import { InteractBody, TurnBody } from './ingest-client';
+import { InteractionBody, Payload, TraceBody } from './ingest-client';
 import { AbstractClient } from './utils';
 
-type Payload = SimpleResponse | GoogleRequest | Simple | WebhookResponse;
+const isGoogleRequest = (p: Payload): p is GoogleRequest => (p ? 'type' in p : false);
```
Comment (jonahsnider) [inline lib/clients/analytics.ts:11]:
```suggestion
const isGoogleRequest = (p: Payload): p is GoogleRequest => p && 'type' in p;
```

**Review by jonahsnider — APPROVED** [review]

File: lib/services/dialogflow/lifecycle/es/response.ts (hunk 1 above)
Comment (trs) [reply lib/services/dialogflow/lifecycle/es/response.ts:63]: middlewares.project.attachID

File: lib/clients/analytics.ts (hunk 2 above)
Comment (pmvrmc) [reply lib/clients/analytics.ts]: yeah, leftovers, removed, ty for catching it

**Pull request closed without merging**